Mu2eG4/src/GammaConvGenerator_module.cc

```diff
@@ -0,0 +1,214 @@
+    , verbosity_{conf().verbosity()}
+    , eng_{createEngine(art::ServiceHandle<SeedService>()->getSeed())}
+    , randFlat_{eng_}
+    , useCorrelatedAngleOverKE_(conf().useCorrelatedAngleOverKE())
+    , spectrum_                (new GammaPairConversionSpectrum(&randFlat_, useCorrelatedAngleOverKE_))
+  {
+    produces<mu2e::StageParticleCollection>();
+    process = ProcessCode::mu2eGammaConversion;
+  }
+
+  //================================================================
+  void GammaConvGenerator::produce(art::Event& event) {
+    auto output{std::make_unique<StageParticleCollection>()};
+
+    const auto simh = event.getValidHandle<SimParticleCollection>(simsToken_);
+    const auto gammas = simParticleList(simh,PDGCode::gamma,ProcessCode::conv);
+
+    if(gammas.empty()) {
+      throw   cet::exception("NO GAMMAS")
+        <<"GammaConvGenerator::produce(): no suitable converted gamma in the input SimParticleCollection\n";
+    }
+
+    // Take the converted gamma as SimParticle
+    const auto gammapart = gammas.at(eng_.operator unsigned int() % gammas.size());
+    const auto p4_gamma = gammapart->endMomentum();
+    const auto t_gammastop = gammapart->endGlobalTime();
+    if (verbosity_>2){
+      std::cout << "GAMMA 4-MOMENTUM: " << p4_gamma << std::endl;
+      std::cout << "END VOLUME INDEX: " << gammapart->endVolumeIndex() << std::endl;
+    }
+    //
+    // Handle to information about G4 physical volumes.
+    bool PhysicalVolumeInfoFound=false;
+    std::string wantedbranch="mu2e::PhysicalVolumeInfomvs_"+physVolModuleLabel_+"_eventlevel_"+physVolProcessLabel_+".";
+    if (verbosity_>2) std::cout << "Searching for " << wantedbranch << std::endl;
+
+    std::vector< art::Handle<PhysicalVolumeInfoMultiCollection> > vah = event.getMany<PhysicalVolumeInfoMultiCollection>();
+    art::Handle<PhysicalVolumeInfoMultiCollection> PhysicalVolumeInfos;
+
+    for (auto const & ah : vah){
+      std::string branchName = ah.provenance()->productDescription().branchName();
+      if (verbosity_>2) std::cout << "PHYSICALVOLUMEINFO COLLECTION: " << branchName << std::endl;
+      if (branchName==wantedbranch){
+        PhysicalVolumeInfoFound=true;
+        PhysicalVolumeInfos=ah;
+        if (verbosity_>2){
+          std::cout << "PHYSICALVOLUMEINFO COLLECTION FOUND" << std::endl;
+          std::cout << PhysicalVolumeInfos << std::endl;
+        }
+      }
+    }
+    if (!PhysicalVolumeInfoFound){
+      throw   cet::exception("NO PHYSICALVOLUMEINFO COLLECTION")
+        <<"GammaConvGenerator::produce(): no suitable PhysicalVolume collection\n";
+    }
+    if (PhysicalVolumeInfos->empty()){
+      throw   cet::exception("EMPTY PHYSICALVOLUMEINFO COLLECTION")
+        <<"GammaConvGenerator::produce(): PhysicalVolume collection is empty\n";
+    }
+    //const auto& vols = PhysicalVolumeInfos->at(gammapart->simStage());
+    /*
+    PhysicalVolumeInfoSingleStage const* vols = &PhysicalVolumeInfos->at(gammapart->simStage());
```

Comment:
Please either remove the commented out code or put it behind a verbosity level.  The one exception to this request is if the commented out code is something in development.

Reply:
removed